crates/pallet-domains/src/domain_registry.rs

```diff
@@ -71,10 +71,14 @@ pub struct DomainConfig<AccountId: Ord, Balance> {
     /// The probability of successful bundle in a slot (active slots coefficient). This defines the
     /// expected bundle production rate, must be `> 0` and `≤ 1`.
     pub bundle_slot_probability: (u64, u64),
-    /// Allowed operators to operate for this domain.
+    /// Accounts allowed to operate on this domain.
     pub operator_allow_list: OperatorAllowList<AccountId>,
-    // Initial balances for Domain.
+    // Initial balances for this domain.
     pub initial_balances: Vec<(MultiAccountId, Balance)>,
+    /// If this is an EVM domain, accounts initially allowed to create contracts on this domain.
+    /// The domain owner can update this list using a sudo call.
+    pub initial_evm_contract_creation_allow_list:
+        Option<PermissionedActionAllowedBy<EthereumAccountId>>,
```

Comment:
instead of adding this field directly to Config, I would prefer if we have an enum for Specific Runtimes starting with just EVM. This will allow use to easily add more params if required in the future. 

Also agree that Option can be removed if Default is AllowedAll

Reply:
I proposed `Option<PermissionedActionAllowedBy<EthereumAccountId>>` because this config item is only needed for evm domain, for other types of domains (e.g. auto-id domain) they should not need to set this config item. It makes sense to me to remove `Option` if we have a domain-specific config.